### PR TITLE
Serbia

### PR DESCRIPTION
**Pull request by Vesihiisi**
Mapping: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/rs_(sr)
Preview: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/rs_(sr)/preview

Task: https://phabricator.wikimedia.org/T179812

### CODE REVIEW

File: importer/RsSr.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        self.set_image()
+        self.set_commonscat()
+        self.set_wd_item(self.find_matching_wikidata(mapping))
+
+
+if __name__ == "__main__":
+    """Point of entrance for importer."""
+    args = importer.handle_args()
+    dataset = Dataset("rs", "sr", RsSr)
+    dataset.data_files = {"admin": "serbia_admin.json",
```
Comment (lokal-profil) [inline importer/RsSr.py]: Not added to the patch yet

File: importer/RsSr.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            self.add_to_report("district", self.district, "located_adm")
+
+    def update_descriptions(self):
+        descs = {
+            "en": "protected cultural monument in Serbia",
+            "nl": "beschermd monument in Servië",
+            "ru": "охраняемый памятник культуры в Сербии",
+            "sr": "заштићен споменик културе у Србији",
+        }
+        for lang in descs:
```
Comment (lokal-profil) [inline importer/RsSr.py]:
```python
for lang, desc in descs.items():
    self.add_description(lang, desc)
```

File: importer/RsSr.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        adm_q = None
+        adm_dic = self.data_files["admin"]
+        okr_dic = self.data_files["okruzi"]
+
+        municip_try = utils.get_item_from_dict_by_key(
+            dict_name=adm_dic,
+            search_in="itemLabel",
+            search_term=self.district)
+        if len(municip_try) == 1:
+            adm_q = municip_try[0]
```
Comment (lokal-profil) [inline importer/RsSr.py]: `else: self.add_to_report("district", self.district, "located_adm")` (if seld.district is never empty, else might want to use `elif self.dirstrict`

File: importer/RsSr.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+        if not adm_q:
+            iso_try = utils.get_item_from_dict_by_key(
+                dict_name=okr_dic,
+                search_in="iso",
+                search_term=self.iso_okrug)
+            if len(iso_try) == 1:
+                adm_q = iso_try[0]
```
Comment (lokal-profil) [inline importer/RsSr.py]: `else: self.add_to_report("iso_okrug", self.iso_okrug, "located_adm")` (if seld.iso_okrug is never empty, else might want to use `elif self.iso_okrug`

File: importer/RsSr.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            if self.has_non_empty_attribute("city"):
+                if utils.contains_digit(self.address):
+                    city_no_brackets = utils.get_rid_of_brackets(self.city)
+                    address = "{}, {}".format(self.address, city_no_brackets)
+                    self.add_statement("located_street", address)
+                else:
+                    dirs = utils.package_monolingual("sr", self.address)
+                    self.add_statement("directions", dirs)
+
+    def set_location(self):
```
Comment (lokal-profil) [inline importer/RsSr.py]: Could replace by 
```python
set_from_dict_match(
    lookup_dict=self.data_files["settlements"],
    dict_label="itemLabel",
    value_label="city",
    prop="location")
```

File: importer/RsSr.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    def set_address(self):
+        """
+        Set address / directions.
+
+        Address, with attached city, if there's digits in it.
+        Otherwise directions.
+        """
+        if self.has_non_empty_attribute("address"):
+            if self.has_non_empty_attribute("city"):
```
Comment (lokal-profil) [inline importer/RsSr.py]: Do we not want to add directions if there is no city?

File: importer/RsSr.py (hunk 2 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: Done.

File: importer/RsSr.py (hunk 1 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: Done.

File: importer/RsSr.py (hunk 3 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: self.district is never empty. done.

File: importer/RsSr.py (hunk 4 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: iso_okrug is empy in many cases so I reworded this.

File: importer/RsSr.py (hunk 5 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: Done.

File: importer/RsSr.py (hunk 6 above)
Comment (Vesihiisi) [reply importer/RsSr.py]: Added.

**Comment by Vesihiisi** [comment]
Preview updated here.

**Pull request merged**